src/prpy/tsr/tsr.py

```diff
@@ -231,14 +334,113 @@ def from_yaml(x, *args, **kw_args):
         x_dict = yaml.safe_load(x, *args, **kw_args)
         return TSR.from_dict(x_dict)
 
-    def sample(self):
-        if len(self.TSRs) == 0:
-            return None
+    def is_valid(self, xyzrpy, ignoreNAN=False):
+        """
+        Checks if a xyzrpy list is a valid sample from the TSR.
+        @param xyzrpy a list of xyzrpy values
+        @param ignoreNAN (optional, defaults to False) ignore NaN xyzrpy
+        @return True if valid and False if not
+        """
+
+        if len(xyzrpy) != len(self.TSRs):
+            raise('Sample must be of equal length to TSR chain!')
+
+        for idx in range(len(self.TSRs)):
+            if not self.TSRs[idx].is_valid(xyzrpy[idx], ignoreNAN):
+                return False
+
+        return True
```

Comment:
The advantage of explicitly looping is the small gain in efficiency and the perhaps larger gain in code clarity? :)